Sources/SwiftKuery/ConnectionPool.swift

```diff
@@ -166,6 +167,29 @@ public class ConnectionPool {
         unlockPoolLock()
     }
     
+    private func increasePoolSize() -> Connection? {
+        defer {
+            unlockPoolLock()
+        }
+        lockPoolLock()
+        var connection: Connection? = nil
+        if capacity == 0 {
+            connection = populateConnectionPool()
+        }
+        return connection
+    }
+    
+    private func populateConnectionPool() -> Connection? {
```

Comment:
This should not be adding a new item to the pool _and_ returning it, it should do one or the other.  It's not valid to `take()` a connection that is still in the pool, as that means two threads could get hold of it, and it would be `give()`n back twice.